pkg/apis/kubermatic/v1/external_cluster.go

```diff
@@ -74,47 +97,30 @@ type ExternalClusterCloudSpec struct {
 	KubeOne *ExternalClusterKubeOneCloudSpec `json:"kubeone,omitempty"`
 }
 
-type Status string
+type ConditionStatus string
```

Comment:
Can we call this `ExternalClusterCondition`? `ConditionStatus` is both unspecific (yet the constants only make sense for external clusters) and overly specific (like "dead corpse").

Maybe even "phase" instead of "status", as that's what we're using for clusters and seeds already. Plus it wouldn't collide with the pre-existing concept of Conditions and their statusses. So maybe just `ExternalClusterPhase`?